api/opencl_runtime_layer.asciidoc

```diff
@@ -304,19 +392,49 @@ Otherwise, it returns one of the following errors:
     by the OpenCL implementation on the device.
   * {CL_OUT_OF_HOST_MEMORY} if there is a failure to allocate resources
     required by the OpenCL implementation on the host.
+--
+
+[open,refpage='clSetCommandQueueProperty',desc='Enable or disable properties of a command-queue.',type='protos']
+--
+To enable or disable the properties of a command-queue, call the function
+
+include::{generated}/api/protos/clSetCommandQueueProperty.txt[]
```

Comment:
The description of this API was transferred faithfully from the older version of the OpenCL spec, but I think we should add a stronger warning that it may not behave as expected in newer versions of OpenCL.  For example, the unified headers currently has the comment:

```c
    /*
     *  WARNING:
     *     This API introduces mutable state into the OpenCL implementation. It has been REMOVED
     *  to better facilitate thread safety.  The 1.0 API is not thread safe. It is not tested by the
     *  OpenCL 1.1 conformance test, and consequently may not work or may not work dependably.
     *  It is likely to be non-performant. Use of this API is not advised. Use at your own risk.
     *
     *  Software developers previously relying on this API are instructed to set the command queue
     *  properties when creating the queue, instead.
     */
```

Perhaps this text should also be included in the spec, after the "NOTE" below?

Reply:
Thanks, this is helpful.   Handling thread safety language is still on the TODO list, in particular due to the differences between 1.0 and 1.1+.

Reply:
I've filed this as an issue so that I could look at the thread-safety descriptions all at once: https://github.com/KhronosGroup/OpenCL-Docs/issues/155